ci-operator/step-registry/cucushift/installer/rehearse/gcp/cases/valid-confidential-computing/provision/cucushift-installer-rehearse-gcp-cases-valid-confidential-computing-provision-commands.sh

```diff
@@ -0,0 +1,320 @@
+
+  supported_cc_types=${instance_type_supported_cc_type_dict["${instance_type}"]}
+  unsupported_cc_types=${instance_type_unsupported_cc_type_dict["${instance_type}"]}
+
+  if ${use_default_flag}; then
+    cc_type1="${default_machine_platform_cc_type}"
+    cc_type2="${default_machine_platform_cc_type}"
+  else
+    cc_type1="${control_plane_cc_type}"
+    cc_type2="${compute_cc_type}"
+  fi
+
+  expected_err_msg=""
+  if [[ "${on_host_maintenance}" != "Terminate" ]]; then
+    expected_err_msg="controlPlane.platform.gcp.onHostMaintenance: Invalid value: \"${on_host_maintenance}\": onHostMaintenace must be set to Terminate when confidentialCompute is ${cc_type1}"
+  fi
+
+  set +o errexit
+  if echo "${unsupported_cc_types}" | grep -qw "${cc_type1}"; then
+    if [[ "${expected_err_msg}" != "" ]]; then
+      expected_err_msg="${expected_err_msg}, "
+    fi
+    expected_err_msg="${expected_err_msg}controlPlane.platform.gcp.type: Invalid value: \"${instance_type}\": Machine type do not support ${cc_type1}. Machine types supporting ${cc_type1}: ${cc_type_machine_series_dict[${cc_type1}]}"
+  fi
+
+  if [[ "${on_host_maintenance}" != "Terminate" ]]; then
+    expected_err_msg="${expected_err_msg}, compute[0].platform.gcp.onHostMaintenance: Invalid value: \"${on_host_maintenance}\": onHostMaintenace must be set to Terminate when confidentialCompute is ${cc_type2}"
+  fi
+
+  if echo "${unsupported_cc_types}" | grep -qw "${cc_type2}"; then
+    if [[ "${expected_err_msg}" != "" ]]; then
+      expected_err_msg="${expected_err_msg}, "
+    fi
+    expected_err_msg="${expected_err_msg}compute[0].platform.gcp.type: Invalid value: \"${instance_type}\": Machine type do not support ${cc_type2}. Machine types supporting ${cc_type2}: ${cc_type_machine_series_dict[${cc_type2}]}"
+  fi
+  set -o errexit
+}
+
+function random_choice()
+{
+  local -r cc_types_str="$1"
+  local array_len selected_index
+
+  IFS=' ' read -r -a cc_types_array <<< "${cc_types_str}"
+  array_len="${#cc_types_array[@]}"
+  selected_index=$(( $RANDOM%($array_len) ))
+  echo "${cc_types_array[${selected_index}]}"
+}
+
+function create_manifests()
+{
+  local -r instance_type="$1"
+  local -r cc_types_str="$2"
+  local -r on_host_maintenance="$3"
+  local -r testing_scenario_num="$4"
+  local -r success_flag="$5"
+  local cluster_name install_dir compute_cc_type control_plane_cc_type default_machine_platform_cc_type result tmp_output use_default_flag rand_int
+
+  tmp_output=$(mktemp)
+  cluster_name="${CLUSTER_PREFIX}${testing_scenario_num}"
+  install_dir="/tmp/${cluster_name}"
+  mkdir -p "${install_dir}" 2>/dev/null
+  create_install_config "${cluster_name}" "${install_dir}"
+
+  compute_cc_type=$(random_choice "${cc_types_str}")
+  control_plane_cc_type=$(random_choice "${cc_types_str}")
+  default_machine_platform_cc_type=$(random_choice "${cc_types_str}")
+
+  use_default_flag=false
+  rand_int=$(( $RANDOM%100 ))
+  if [[ $rand_int -gt 30 ]]; then
```

Comment:
From the following matrix, there are 3 combination:
For now, for all kinds of combination matrix, I can figure out:
1. only `defaultMachinePlatform` configured.
2. both worker and controlPlane configured
3.  worker, controlPlane and defaultMachinePlatform configured.

So can we use `rand_int=$(( $RANDOM%3 ))` to map the 3 cases, then we do not have to introduce `use_default_flag` ?

Reply:
Do you mean below? 
```
rand_int=$(( $RANDOM%3 ))
if [[ $rand_int -eq 0 ]]; then
  # 1. only defaultMachinePlatform configured
elif [[ $rand_int -eq 1 ]]; then
  # 2. both worker and controlPlane configured
else
  # 3. worker, controlPlane and defaultMachinePlatform configured
fi

Reply:
right